src/main/apiClient.ts

```diff
@@ -600,6 +603,33 @@ export default class ApiClient {
     }
   }
 
+  addNewWebAuthnDevice(accessToken: string, auth?: AuthOptions): Promise<void> {
+    return this.http
+      .post<PublicKeyCredentialCreationOptionsSerialized>('/webauthn/makeCredentials', { accessToken })
+      .then(serialiedOptions => {
+        const publicKey = encodePublicKeyCredentialCreationOptions(serialiedOptions)
+
+        return navigator.credentials.create({ publicKey })
+      })
+      .then(credentials => {
+        if (!credentials || credentials.type !== publicKeyCredentialType) {
+          throw new Error('Unable to register invalid public key crendentials.')
+        }
+
+        const serializedCredentials = serializePublicKeyCredential(credentials)
+
+        return this.http
+          .post<InternalToken>('/webauthn/credentials', { body: { ...serializedCredentials }, accessToken })
+          .then(token => this.loginWithPasswordCallback(token.tkn, auth))
```

Comment:
`loginWithPasswordCallback` does not need to be called. `/webauthn/credentials` will return a 2XX if the operation succeed otherwise it will return an error message.